src/oracles/ChainlinkPriceOracle.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.0 <0.9.0;
+
+import {AggregatorV3Interface} from "../interfaces/AggregatorV3Interface.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+
+/**
+ * @title Chainlink Price Oracle
+ * @author GUNBOATs
+ * @dev This contract creates an oracle that is compatible with the IPriceOracle
+ * interface and can be used by a CoW AMM to determine the current price of the
+ * traded tokens from Chainlink Data Feeds with decimals lower than 18.
+ */
+contract ChainlinkPriceOracle is IPriceOracle {
+    /**
+    * @param token0Feed Address of token0 oracle
+    * @param token1Feed Address of token1 oracle
+    * @param timeThreshold amount of seconds before it consider as stale oracle
+    */
+    struct Data {
+        address token0Feed;
+        address token1Feed;
+        uint256 timeThreshold;
+    }
+    /**
+     * @dev There is no mapping between Chainlink oracle and the token address,
+     * so the user must check the that the oracle is conresponded to the token by themself.
+     * @inheritdoc IPriceOracle
+     */
+    function getPrice(address, address, bytes calldata data)
+        external
+        view
+        returns (uint256 priceNumerator, uint256 priceDenominator)
+    {
+        Data memory OracleData = abi.decode(data, (Data));
+        AggregatorV3Interface token0Feed = AggregatorV3Interface(OracleData.token0Feed);
+        AggregatorV3Interface token1Feed = AggregatorV3Interface(OracleData.token1Feed);
+        ( /* uint80 roundId*/
+            , int256 token0Answer, /* uint256 startedAt */, uint256 token0Timestamp, /* uint80 answerInRound */
+        ) = token0Feed.latestRoundData();
+        ( /* uint80 roundId*/
+            , int256 token1Answer, /* uint256 startedAt */, uint256 token1Timestamp, /* uint80 answerInRound */
```

Comment:
nit: it's nice that the unused components of the return data from `latestRoundData` are commented so it can be seen, however IMO this reduces readability (and perhaps leads to preselection bias for auditors / readers of the code). For these reasons, I think comments should be removed here, or new lines used for each parameter so it is easy to see _what_ is used clearly and where.